admin/automation/sync.py

```diff
@@ -0,0 +1,231 @@
+            "-p",
+            "--projects",
+            help="Number of projects to process",
+            default=1,
+            type=int,
+        )
+        self.parser.add_argument(
+            "--repo",
+            help="Repository where the automation should happen",
+            default="ngipkgs",
+        )
+        self.parser.add_argument(
+            "--credentials",
+            help="Directory from which GitHub credentials are loaded",
+            type=dir_path,
+            default="./.env",
+        )
+        self.parser.add_argument(
+            "--template",
+            help="Directory for the project template",
+            type=dir_path,
+            default="./template",
+        )
+
+        self.args = self.parser.parse_args()
+
+
+def dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def load_credentials(directory):
+    for filename in os.listdir(directory):
+        file_path = os.path.join(directory, filename)
+
+        if os.path.isfile(file_path):
+            with open(file_path, "r") as file:
+                content = file.read().strip()
+                os.environ[filename] = content
+
+
+class Deliverable(str, Enum):
+    SERVICES = "services"
+    EXECUTABLES = "executables"
+    LIBRARIES = "libraries"
+    TESTS = "tests"
+    DEVENV = "development environment"
+
+
+@dataclass
+class Project:
+    name: str
+    branch_name: str = field(init=False)
+    description: str = "\n### Websites"
+    websites: Series = field(default_factory=Series)
+
+    def __post_init__(self):
+        self.branch_name = f"projects/{self.name}"
+
+    def append_websites(self, websites: List[str]):
+        self.websites = pd.concat(
+            [self.websites, pd.Series(websites)], ignore_index=True
+        )
+
+    # TODO: make a template for this
+    def update_description(self):
+        if len(self.websites) > 0:
+            pass
+
+        for site in self.websites:
+            self.description += f"\n- {site}"
+
+    def clean_websites(self):
+        """Remove empty values, clean up URLs and remove duplicates"""
+        self.websites = cleanup_empty(self.websites)
+        self.websites = cleanup_urls(self.websites)
+        self.websites = self.websites.drop_duplicates()
+
+
+class Subgrants(BaseModel):
+    subgrants: List[Subgrant] = []
+
+    def get_websites(self, name: str) -> List[str]:
+        for s in self.subgrants:
+            if name == s.name:
+                return s.websites
+        return []
+
+
+class NotionProject(BaseModel):
+    name: str = ""
+    subgrants: List[str] = []
+
+
+# TODO: get status from Notion
+def main():
+    count = 1
+
+    projects = pd.read_csv(input_file, usecols=["Name", "Subgrants"])
+    projects["Name"] = cleanup_empty(projects["Name"])
+
+    # Some project names have spaces in them, so we'd either need to remove
+    # that in Notion or just replace them here.
```

Comment:
Hm, why is it a problem here that those names have spaces? Those are the project names, they're fixed and we can't just mangle them.

Reply:
The only problem is that we can't have spaces in branch names. Maybe it's better to only do this when creating branches.

Reply:
Yes of course, only for branches.